.github/workflows/ci.yml

```diff
@@ -1,32 +1,37 @@
 name: ci
 
-on: [push, pull_request]
+on: [ push, pull_request ]
 
 jobs:
 
   build:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.9", "3.10", "3.11", "3.12"]
+        python-version: [ "3.9", "3.10", "3.11", "3.12" ]
 
     steps:
-    - uses: actions/checkout@v4
+      - uses: actions/checkout@v4
 
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v5
-      with:
-        python-version: ${{ matrix.python-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
 
-    - name: Install poetry
-      run: pipx install poetry
+      - name: Install poetry
+        run: pipx install poetry
 
-    - name: Install python dependencies
-      run: |
-        poetry env use ${{ matrix.python-version }}
-        poetry install --with dev
+      - name: Install python dependencies
+        run: |
+          poetry env use ${{ matrix.python-version }}
+          poetry install --with dev
 
-    - name: Run unit tests
-      run: |
-        poetry run python -m unittest -v tests
+      - name: Run unit tests
+        run: |
+          poetry run pytest --cov=abcd --cov-report xml --cov-report term:skip-covered
+
+      - name: Upload coverage reports to Codecov
+        uses: codecov/codecov-action@v4.0.1
```

Comment:
Any reason for pinning to 4.0.1 specifically? I think the most recent release is 4.4.1.

Reply:
no special intention, the codecov instructions provided this snippet